server/backend/mlx_runner.py

```
@@ -491,27 +496,45 @@ def generate_streaming_gpt(
-            if is_analysis is None and parser.current_channel == "analysis":
-                is_analysis = True
+            # if (
+            #     parser.current_channel == "analysis"
+            #     and parser.current_recipient != None
+            # ):
+            #     print("FOUND TOOL CALL IN REASONING ANOMALY")
+
+            if (
+                current_channel != "analysis"
+                and parser.current_channel == "analysis"
+                and parser.current_recipient == None
```

Ruff (0.15.17)

[error] Comparison to `None` should be `cond is None`

Replace with `cond is None`

(E711)